qctrlopencontrols/driven_controls/driven_control.py

```diff
@@ -545,42 +497,62 @@ def get_plot_formatted_arrays(self, coordinates=CARTESIAN, dimensionless_rabi_ra
         ArgumentsValueError
             Raised when an argument is invalid.
         """
+        if coordinates not in [CARTESIAN, CYLINDRICAL]:
+            raise ArgumentsValueError(
+                'Unsupported coordinates provided: ',
+                arguments={'coordinates': coordinates})
+
         if dimensionless_rabi_rate:
             normalizer = self.maximum_rabi_rate
         else:
             normalizer = 1
 
         if coordinates == CARTESIAN:
-            (x_amplitudes, y_amplitudes, detunings, times) = get_plot_data_from_segments(
-                np.vstack((self.amplitude_x / normalizer, self.amplitude_y / normalizer,
-                           self.detunings, self.durations)).T
-            )
-            plot_dictionary = {
-                'amplitudes_x': x_amplitudes,
-                'amplitudes_y': y_amplitudes,
-                'detunings': detunings,
-                'times': times
-            }
-
+            control_segments = np.vstack((
+                self.amplitude_x / normalizer,
+                self.amplitude_y / normalizer,
+                self.detunings,
+                self.durations)).T
         elif coordinates == CYLINDRICAL:
-            (x_plot, y_plot, detunings, times) = get_plot_data_from_segments(
-                np.vstack((self.rabi_rates / normalizer, self.azimuthal_angles,
-                           self.detunings, self.durations)).T
-            )
+            control_segments = np.vstack((
+                self.rabi_rates / normalizer,
+                self.azimuthal_angles,
+                self.detunings,
+                self.durations)).T
+
+        segment_times = np.insert(np.cumsum(control_segments[:, 3]), 0, 0.)
+        plot_time = (segment_times[:, np.newaxis] * np.ones((1, 2))).flatten()
+        plot_amplitude_x = control_segments[:, 0]
+        plot_amplitude_y = control_segments[:, 1]
+        plot_amplitude_z = control_segments[:, 2]
+
+        plot_amplitude_x = np.concatenate(
+            ([0.], (plot_amplitude_x[:, np.newaxis] * np.ones((1, 2))).flatten(), [0.]))
+        plot_amplitude_y = np.concatenate(
+            ([0.], (plot_amplitude_y[:, np.newaxis] * np.ones((1, 2))).flatten(), [0.]))
+        plot_amplitude_z = np.concatenate(
+            ([0.], (plot_amplitude_z[:, np.newaxis] * np.ones((1, 2))).flatten(), [0.]))
+
+        plot_dictionary = {
```

Comment:
IMO it'd be a bit clearer to put this in an "if coordinates == CARTESIAN" block, so we're only setting plot_dictionary once (alternatively, we could just return the dictionary directly from the two if blocks, there's no need for the local variable at all).

Reply:
@charmasaur - I agree. Linter objects to some other ways. Just updated a method that may be a bit clearer and linter friendly.